test/test_functionalization.py

```diff
@@ -25,27 +25,35 @@ def are_aliased(x, y):
 
 # We can unify testing and use functionalize() here instead
 # if/when functorch moves into core.
-# This is basically a crappy version of `functionalize()` for single-tensor-arg inputs.
+# This is basically a crappy version of `functionalize()`.
 def _functionalize(f, *, reapply_views: bool, crossref: bool):
-    def wrapped(a):
+    def to_fun(t: torch.Tensor):
+        func_t = torch._to_functional_tensor(t)
+        func_t.requires_grad = t.requires_grad
+        return func_t
+
+    def wrapped(*inputs):
         ctx = nullcontext()
         if crossref:
             ctx = enable_crossref_functionalize()
         with ctx:
-            input_functional = torch._to_functional_tensor(a)
-            input_functional.requires_grad = a.requires_grad
+            inputs_functional = tree_map_only(torch.Tensor, to_fun, inputs)
             torch._enable_functionalization(reapply_views=reapply_views)
+            flat_inputs, _ = tree_flatten(inputs)
+            flat_inputs_functional, _ = tree_flatten(inputs_functional)
             try:
-                out = f(input_functional)
+
+                out = f(*flat_inputs_functional)
```

Comment:
nit: I think we'll want to use `inputs_functional` here instead of `flat_inputs_functional`

Specifically if we have something like
```
def f(x: List[tuple[Tensor]]):
   ...
```
and we then end up passing the wrapper function a `List[tuple[Tensor]]`, this will end up passing `f` a `List[Tensor]`

Reply:
good catch!